plugins/modules/azure_rm_appgateway.py

```diff
@@ -2291,7 +2293,8 @@ def exec_module(self, **kwargs):
                     not compare_arrays(old_response, self.parameters, 'url_path_maps') or
                     not compare_arrays(old_response, self.parameters, 'trusted_root_certificates') or
                     not compare_dicts(old_response, self.parameters, 'autoscale_configuration') or
-                    not compare_dicts(old_response, self.parameters, 'web_application_firewall_configuration')):
+                    not compare_dicts(old_response, self.parameters, 'web_application_firewall_configuration') or
+                    not compare_dicts(old_response, self.parameters, 'tags')):
```

Comment:
@jmighion If you judge tags this way, it will be better, so that you can keep the existing tags. Thank you!
update_tags, new_tags = self.update_tags(old_response['tags'])
if update_tags:
    self.to_do = Actions.Update
    self.parameters["tags"] = new_tags

Reply:
Not seeing where you'd like that to go. Completely separate conditional after this one?

Reply:
@jmighion Yes. Separate conditional judgment, get merged tags. Thank you!

Reply:
Cool cool. Have a look!

Reply:
@jmighion Thank you for the update, I will test and move forward with the merge as soon as possible.